conans/client/rest/rest_client.py

```diff
@@ -18,7 +22,12 @@ def __init__(self, output, requester, put_headers=None):
         self.remote_url = None
         self.custom_headers = {}  # Can set custom headers to each request
         self._output = output
-        self.requester = requester
+
+        # Verify client version against remotes
+        self.requester = VersionCheckerRequester(requester, Version(client_version),
```

Comment:
I think this issue meant to completely remove ``VersionCheckerRequester``, not move it to a different location. The versions are no longer relevant, not checked in practice. Remove the component.

Reply:
Ok, so there are a lot of things to remove 🚧 🏗 👷